velox/connectors/hive/HiveDataSource.cpp

```diff
@@ -585,20 +585,7 @@ void HiveDataSource::addSplit(std::shared_ptr<ConnectorSplit> split) {
     auto fieldName = readerOutputType_->nameOf(i);
     auto scanChildSpec = scanSpec_->childByName(fieldName);
 
-    auto keyIt = split_->partitionKeys.find(fieldName);
-    if (keyIt != split_->partitionKeys.end()) {
-      setPartitionValue(scanChildSpec, fieldName, keyIt->second);
-    } else if (fieldName == kPath) {
-      setConstantValue(
-          scanChildSpec, VARCHAR(), velox::variant(split_->filePath));
-    } else if (fieldName == kBucket) {
-      if (split_->tableBucketNumber.has_value()) {
-        setConstantValue(
-            scanChildSpec,
-            INTEGER(),
-            velox::variant(split_->tableBucketNumber.value()));
-      }
-    } else if (!fileType->containsChild(fieldName)) {
+    if (!fileType->containsChild(fieldName)) {
```

Comment:
Just merge the 3 in an empty branch and do nothing in these cases?  this way we can avoid allocating null constants then overwrite them later.  Or comment that they will be overwritten later here.

Reply:
Thanks for brainstorming. Currently, the loop is on every output column (readerOutputType_), but the columns with filters may not be output columns. Also, scanSpec_->childByName(entry.first); is an O(n) operation. How about looping over scanSpec_ children which contains both? This way we will have the loop once, and we will be able to remove the scanSpec_->childByName calls. I'll try it and update the PR tomorrow.